src/NServiceBus.Core/Transports/Msmq/CheckEndpointNameComplianceForMsmq.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace NServiceBus.Transports.Msmq
+{
+    using System;
+
+    class CheckEndpointNameComplianceForMsmq
+    {
+        public void Check(string endpointName)
+        {
+            // .NET Messaging API hardcodes the buffer size to 124. As a result, the entire format name of the queue cannot exceed 123
+            var formatName = $"DIRECT=OS:{Environment.MachineName}\\private$\\{endpointName}";
+            if (formatName.Length > 123)
+            {
+                throw new InvalidOperationException($"The specified endpoint name {endpointName} is too long. The format name for the endpoint:{formatName} needs to be 123 characters or less.");
```

Comment:
@Particular/msmq-transport-maintainers - I think this exception message needs to be changed a little. As it's the queue name and not the endpoint name. However the sub queue doesn't matter, as the endpoint name is what needs to be shortened. Any suggestions on how we can improve the message?